manage_user_reset.php

```diff
@@ -52,33 +52,37 @@
 );
 
 $t_command = new UserResetPasswordCommand( $t_data );
-$t_command->execute();
+$t_result = $t_command->execute();
+$t_result = $t_result['result'];
 
 $t_redirect_url = 'manage_user_page.php';
 
 form_security_purge( 'manage_user_reset' );
 
-layout_page_header( null, $t_result ? $t_redirect_url : null );
-
+layout_page_header(
+	null,
+	$t_result != UserResetPasswordCommand::RESULT_FAILURE ? $t_redirect_url : null
+);
 layout_page_begin( 'manage_overview_page.php' );
 
-if( $t_reset ) {
-	if( false == $t_result ) {
-		# PROTECTED
-		html_operation_failure( $t_redirect_url, lang_get( 'account_reset_protected_msg' ) );
-	} else {
-		# SUCCESSFUL RESET
-		if( ( ON == config_get( 'send_reset_password' ) ) && ( ON == config_get( 'enable_email_notification' ) ) ) {
-			# send the new random password via email
+switch( $t_result ) {
+	case UserResetPasswordCommand::RESULT_RESET:
+		if(    ( ON == config_get( 'send_reset_password' ) )
+			&& ( ON == config_get( 'enable_email_notification' ) )
+		) {
+			# Password reset confirmation sent by email
 			html_operation_successful( $t_redirect_url, lang_get( 'account_reset_msg' ) );
 		} else {
-			# email notification disabled, then set the password to blank
+			# Email notification disabled, password set to blank
 			html_operation_successful( $t_redirect_url, lang_get( 'account_reset_msg2' ) );
 		}
-	}
-} else {
-	# UNLOCK
-	html_operation_successful( $t_redirect_url, lang_get( 'account_unlock_msg' ) );
+		break;
+	case UserResetPasswordCommand::RESULT_UNLOCK:
+		html_operation_successful( $t_redirect_url, lang_get( 'account_unlock_msg' ) );
+		break;
+	case UserResetPasswordCommand::RESULT_FAILURE:
```

Comment:
This is no longer defined and we should depend on the exceptions for failures.

Reply:
Already fixed.